storey/sources.py

```diff
@@ -730,212 +754,80 @@ def __init__(
-                                else:
-                                    single_key_field = self._key_field
-                                    if self._with_header and isinstance(single_key_field, str):
-                                        single_key_field_index = field_name_to_index[single_key_field]
-                                    else:
-                                        single_key_field_index = single_key_field
-                                    key = parsed_line[single_key_field_index]
-                                    if key is None:
-                                        create_event = False
-                            if create_event:
-                                if self._id_field:
-                                    id_field = self._id_field
-                                    if self._with_header and isinstance(id_field, str):
-                                        id_field = field_name_to_index[id_field]
-                                    id = parsed_line[id_field]
-                                else:
-                                    id = self._get_uuid()
-                                event = Event(element, key=key, id=id)
-                                self._event_buffer.put(event)
-                            else:
-                                if self.context:
-                                    self.context.logger.error(
-                                        f"For {parsed_line} value of key {single_key_field} is None"
-                                    )
-                        except BaseException as ex:
-                            ex._raised_by_storey_step = self
-                            if self.context and hasattr(self.context, "push_error"):
-                                message = traceback.format_exc()
-                                if self.logger:
-                                    self.logger.error(f"Pushing error to error stream: {ex}\n{message}")
-                                if not event:
-                                    if parsed_line is not None:
-                                        event = Event(parsed_line)
-                                    else:
-                                        event = Event(line)
-                                self.context.push_error(event, f"{ex}\n{message}", source=self.name)
-                            else:
-                                raise ex
-                if self._with_header:
-                    self._dates_indices = []
-
-        except BaseException as ex:
-            self._event_buffer.put(ex)
-        self._event_buffer.put(_termination_obj)
-
-    def _get_event(self):
-        event = self._event_buffer.get()
-        if isinstance(event, BaseException):
-            raise event
-        return event
+    def _get_element(self, body: OrderedDict):
+        if self._build_dict and self._with_header:
```

Comment:
What happens if `self._build_dict and not self._with_header`? We just ignore the `build_dict` setting? That doesn't seem right.

Reply:
You are right, but this what we had in the pervious version so I did not change it.
what do you suggest? we do not have the headers to make it as dict.

Reply:
I guess just make it raise an error in `__init__`.

Reply:
What about this one?